bootstore/src/schemes/v0/fsm.rs

```diff
@@ -0,0 +1,479 @@
+                            our_pkg = Some(pkg.clone());
+                            None
+                        } else {
+                            Some(Envelope {
+                                to: peer,
+                                msg: Request {
+                                    id: request_id,
+                                    type_: RequestType::Init(pkg.clone()),
+                                }
+                                .into(),
+                            })
+                        }
+                    })
+                    .collect();
+                self.state = Some(State::InitialMember(InitialMemberState {
+                    pkg: our_pkg.unwrap(),
+                    distributed_shares: BTreeMap::new(),
+                    rack_init_state: Some(RackInitState::Running {
+                        start: self.common.clock,
+                        total_members,
+                        acks: BTreeSet::from([self.common.id.clone()]),
+                        unacked_envelopes: envelopes.clone(),
+                    }),
+                    pending_learn_requests: BTreeMap::new(),
+                }));
+                // Note that while we could only send the envelopes to
+                // connected peers, the network layer will just drop them.
+                // We already have a copy to send on connection stored in
+                // `Rack::InitState::Running.unacked_envelopes`.
+                Output { persist: true, envelopes, api_output: None }
+            }
+            Err(e) => return ApiError::RackInitFailed(e).into(),
+        }
+    }
+
+    /// Initialize a node added after rack initialization
+    pub fn init_learner(&mut self) -> Output {
+        let State::Uninitialized(_) = self.state.as_ref().unwrap() else {
+            return ApiError::PeerAlreadyInitialized.into();
+        };
+
+        let mut state = LearningState { attempt: None };
+        let mut output = state.new_attempt(&mut self.common);
+        self.state = Some(State::Learning(state).into());
+        // We changed state into a learner, so persist this knowledge
+        output.persist = true;
+        output
+    }
+
+    /// This call is triggered locally after RSS runs, in order to retrieve the
+    /// `RackSecret` so that it can be used as input key material.
+    ///
+    /// if the rack secret has not already been loaded, then share retrieval
+    /// will begin.
+    pub fn load_rack_secret(&mut self) -> Output {
+        let (rack_uuid, local_share, threshold, share_digests) = match self
+            .state
+            .as_ref()
+            .unwrap()
+        {
+            // We don't allow retrieval before initialization
+            State::Uninitialized(_) => {
+                return ApiError::RackNotInitialized.into()
+            }
+            State::Learning(_) => return ApiError::StillLearning.into(),
+            State::InitialMember(InitialMemberState { pkg, .. }) => {
```

Comment:
It seems like cheating to get the rack secret from `InitialMemberState`, since you don't actually have to get the shares from everyone else.  Not sure if it's _wrong_, though...

Reply:
We don't do this as of [98ada27](https://github.com/oxidecomputer/omicron/pull/3385/commits/98ada27b53d4429463cde94cc91bedec0bc4ef98).

Each request loads its own instance of the `RackSecret` via share retreival.